src/components/settings/NumberSetting.tsx

```diff
@@ -62,45 +63,50 @@ export const NumberSetting = ({
     handleUpdate,
     showSlider,
     disabled = false,
+    handleLiveUpdate,
     listItemTextSx: sx,
 }: Props) => {
     const { t } = useTranslation();
 
     const [isDialogOpen, setIsDialogOpen] = useState(false);
     const [dialogValue, setDialogValue] = useState(value);
+    const [originalValue, setOriginalValue] = useState(value);
 
-    const closeDialog = useCallback(
-        (resetValue: boolean) => {
-            setIsDialogOpen(false);
-
-            if (resetValue) {
-                setDialogValue(value);
+    const updateValue = useCallback(
+        (newValue: number) => {
+            setDialogValue(newValue);
+            if (handleLiveUpdate) {
+                handleLiveUpdate(newValue);
             }
         },
-        [value],
+        [handleLiveUpdate],
     );
 
-    const closeDialogWithReset = useCallback(() => closeDialog(true), [closeDialog]);
-
-    const updateSetting = useCallback(
-        (newValue: number, shouldCloseDialog: boolean = true) => {
-            if (shouldCloseDialog) {
-                closeDialog(false);
-            }
+    const cancel = useCallback(() => {
+        setDialogValue(originalValue);
+        setOriginalValue(originalValue);
+        handleUpdate(originalValue);
+        setIsDialogOpen(false);
+    }, [originalValue, handleUpdate]);
 
-            const didValueChange = value !== newValue;
```

Comment:
this should be kept to prevent unnecessary requests, currently clicking cancel sends a requests for the same value